lazy.go

```diff
@@ -119,6 +119,9 @@ func (l *lazyConn) writeHandshake() error {
 }
 
 func (l *lazyConn) Write(b []byte) (int, error) {
+	l.whlock.Lock()
+	defer l.whlock.Unlock()
```

Comment:
My bad. I've removed the lock in `writeHandshake` and added it when calling it in `Read`.